diff -Naur createrepo_c-0.12.1/src/python/CMakeLists.txt createrepo_c-0.12.1.tpg/src/python/CMakeLists.txt
--- createrepo_c-0.12.1/src/python/CMakeLists.txt	2019-02-12 09:45:19.000000000 +0000
+++ createrepo_c-0.12.1.tpg/src/python/CMakeLists.txt	2019-02-28 10:04:26.737000000 +0000
@@ -14,7 +14,7 @@
         FIND_PACKAGE(PythonInterp 2 EXACT REQUIRED)
     elseif (${PYTHON_DESIRED} STREQUAL "3")
         PYTHON_UNSET()
-        SET(Python_ADDITIONAL_VERSIONS 3 CACHE INTERNAL "")
+        SET(Python_ADDITIONAL_VERSIONS 3.0 CACHE INTERNAL "")
         FIND_PACKAGE(PythonLibs 3 EXACT)
         FIND_PACKAGE(PythonInterp 3 EXACT REQUIRED)
     elseif (EXISTS ${PYTHON_DESIRED})
