From f3039c59bb67d595a5969d8c79781c454f6bdc8b Mon Sep 17 00:00:00 2001
From: Neal Gompa <ngompa13@gmail.com>
Date: Sun, 7 Jan 2018 18:57:03 -0500
Subject: [PATCH] Correctly identify Requires(missingok) dependencies as
 Recommends

---
 src/parsehdr.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/parsehdr.c b/src/parsehdr.c
index 2f6a79e..e247618 100644
--- a/src/parsehdr.c
+++ b/src/parsehdr.c
@@ -31,6 +31,8 @@
 #define RPM_WEAK_DEPS_SUPPORT 1
 #endif
 
+#define RPMSENSE_MISSINGOK (1 << 19)
+
 typedef enum DepType_e {
     DEP_PROVIDES,
     DEP_CONFLICTS,
@@ -411,6 +413,10 @@ cr_package_from_header(Header hdr,
                         pkg->obsoletes = g_slist_prepend(pkg->obsoletes, dependency);
                         break;
                     case DEP_REQUIRES:
+                        if ( num_flags & RPMSENSE_MISSINGOK ) {
+                            pkg->recommends = g_slist_prepend(pkg->recommends, dependency);
+                            break;
+                        }
                         dependency->pre = pre;
 
                         // XXX: libc.so filtering ////////////////////////////
-- 
2.15.1

