diff -up createrepo_c-0.21.1/src/createrepo_c.c.omv~ createrepo_c-0.21.1/src/createrepo_c.c
--- createrepo_c-0.21.1/src/createrepo_c.c.omv~	2023-06-11 21:57:05.865975536 +0200
+++ createrepo_c-0.21.1/src/createrepo_c.c	2023-06-11 22:01:55.083988393 +0200
@@ -597,8 +597,14 @@ static int buildtimesort(gconstpointer a
     struct DuplicateLocation *a_loc = (struct DuplicateLocation *)a;
     struct DuplicateLocation *b_loc = (struct DuplicateLocation *)b;
 
-    assert(a_loc->pkg->time_build != 0);
-    assert(b_loc->pkg->time_build != 0);
+    //assert(a_loc->pkg->time_build != 0);
+    //assert(b_loc->pkg->time_build != 0);
+    if(a_loc->pkg->time_build == 0) {
+	    fprintf(stderr, "%s %s-%s: Invalid 0 time_build in %s (while comparing to %s)\n", a_loc->pkg->name, a_loc->pkg->version, a_loc->pkg->release, a_loc->location, b_loc->location);
+    }
+    if(b_loc->pkg->time_build == 0) {
+	    fprintf(stderr, "%s %s-%s: Invalid 0 time_build in %s (while comparing to %s)\n", b_loc->pkg->name, b_loc->pkg->version, b_loc->pkg->release, b_loc->location, a_loc->location);
+    }
 
     // order by build time first
     int64_t result = a_loc->pkg->time_build - b_loc->pkg->time_build;
